src/main/java/org/yorm/util/RowRecordConverter.java

```diff
@@ -1,90 +1,343 @@
+        try {
+            switch (type) {
+                //MySql does not have a truly boolean type, bool/boolean are a synonym of tinyint(1)
+                //Postgresql maps booleans to bits
+                case TINYINT, BIT, BOOLEAN -> preparedStatement.setBoolean(paramIndex, (boolean) value);
+                case SMALLINT, INTEGER -> {
+                    if (value instanceof Integer) {
+                        preparedStatement.setInt(paramIndex, (int) value);
+                    } else if (value instanceof Long) {
+                        preparedStatement.setLong(paramIndex, (long) value);
+                    } else {
+                        throw new YormException("Incompatible value:" + value + " of class:" + value.getClass().getName() + " for column type:" + type);
+                    }
                 }
+                case BIGINT -> preparedStatement.setLong(paramIndex, (long) converter.convert(value));
+                case VARCHAR, TEXT -> preparedStatement.setString(paramIndex, (String) converter.convert(value));
+                case CHAR -> preparedStatement.setByte(paramIndex, (byte) (char) converter.convert(value));
+                case DOUBLE -> preparedStatement.setDouble(paramIndex, (double) converter.convert(value));
+                case FLOAT, REAL -> preparedStatement.setFloat(paramIndex, (float) converter.convert(value));
+                case DECIMAL -> preparedStatement.setBigDecimal(paramIndex, (BigDecimal) converter.convert(value));
+                case DATE -> preparedStatement.setDate(paramIndex, Date.valueOf((LocalDate) converter.convert(value)));
+                case TIME -> preparedStatement.setTime(paramIndex, Time.valueOf((LocalTime) converter.convert(value)));
+                case TIMESTAMP -> preparedStatement.setTimestamp(paramIndex, Timestamp.valueOf((LocalDateTime) converter.convert(value)));
+                default -> throw new YormException("Couldn't find type for " + dbColumnName);
             }
-            case BIGINT -> preparedStatement.setLong(paramIndex, (long) value);
-            case VARCHAR, CHAR, TEXT -> preparedStatement.setString(paramIndex, (String) value);
-            case DOUBLE -> preparedStatement.setDouble(paramIndex, (double) value);
-            case FLOAT, REAL -> preparedStatement.setFloat(paramIndex, (float) value);
-            case DECIMAL -> preparedStatement.setBigDecimal(paramIndex, (BigDecimal) value);
-            case DATE -> preparedStatement.setDate(paramIndex, Date.valueOf((LocalDate) value));
-            case TIME -> preparedStatement.setTime(paramIndex, Time.valueOf((LocalTime) value));
-            case TIMESTAMP -> preparedStatement.setTimestamp(paramIndex, Timestamp.valueOf((LocalDateTime) value));
-            default -> throw new YormException("Couldn't find type for " + dbColumnName);
+        } catch (SQLException e) {
+            throw new RuntimeException(e);
+        } catch (YormException e) {
```

Comment:
These two catch could be combined

Reply:
I'm actually just gonna rip them out! The catch block was useful in debugging...